### DISCUSSION TITLE
First time user feedback

### DISCUSSION THREAD

**Discussion by Adam-D-Lewis** [Quansight/ragna#388] [Ideas]
Hi, I wanted to provide my feedback as a first time user of Ragna.

- As a more advanced user, when going through the ragna init prompt, I would have found it more helpful to say the "not worrying about dependencies" option limits you to dummy use cases only.  I realize the simplicity may be helpful for less technical users however.
- ragna check was not reliable for me when I installed things into a conda environment.  I assume the boxes should have a checkmark in them since I have the packages installed.
![image](https://github.com/Quansight/ragna/assets/23342526/338f9b88-df9a-47f2-bf66-7f3167c44844)
- When I started up my first conversation, even though I selected lancedb, I saw that chroma downloaded an 80 MB file for some reason.
- I wish you could you could hit shift+enter to create a newline without submitting my chat message similar to how you can on ChatGPT website.
- I wish new chats were displayed with the most recent first (highest) rather than oldest first in the rag ui.
- I also wish uploading a document was optional (no RAG) though I realize you might want to specialize in RAG specifically.

**Comment by pmeier** [comment]
Thanks Adam for the feedback. Let me go through the points one by one:

> As a more advanced user, when going through the ragna init prompt, I would have found it more helpful to say the "not worrying about dependencies" option limits you to dummy use cases only. I realize the simplicity may be helpful for less technical users however.

Happy to rephrase:

https://github.com/Quansight/ragna/blob/f7de2e947c04cd3d4e8319cc117a8f9ecfa21239/ragna/deploy/_cli/config.py#L79-L85

Maybe just

> [...] try Ragna with demo components [...]

? If yes, or if you have a better idea, would you like to send a PR for this?

> ragna check was not reliable for me when I installed things into a conda environment.  I assume the boxes should have a checkmark in them since I have the packages installed.

This looks like a font issue on your system. The "check boxes" are probably stand-ins for "missing symbol". Here is how it should look like:

![image](https://github.com/Quansight/ragna/assets/6849766/bd43df93-ee1f-4395-a164-975d3d5ee8d7)

We are using [`rich`](https://github.com/Textualize/rich) and emojis for the checkmarks and crosses:

https://github.com/Quansight/ragna/blob/f7de2e947c04cd3d4e8319cc117a8f9ecfa21239/ragna/deploy/_cli/config.py#L392-L393

I think rich uses a best effort approach to display these. SO either I'm wrong or your terminal doesn't allow any symbols whatsoever. In any case, we should probably look into it. Would you mind creating an issue for this?

> When I started up my first conversation, even though I selected lancedb, I saw that chroma downloaded an 80 MB file for some reason.

That is expected. For now, our source storages need to include the embedding model. And `chromadb` packaged a small one (80MB) instead of requiring a dependency on PyTorch, which alone is multi GB dependency.

https://github.com/Quansight/ragna/blob/f7de2e947c04cd3d4e8319cc117a8f9ecfa21239/ragna/source_storages/_vector_database.py#L55-L62

> I wish you could you could hit shift+enter to create a newline without submitting my chat message similar to how you can on ChatGPT website.

This is related to

https://github.com/Quansight/ragna/blob/f7de2e947c04cd3d4e8319cc117a8f9ecfa21239/ragna/deploy/_ui/central_view.py#L434-L438

IIUC, `panel==1.4.0` fixes this issue. Progress for bumping our dependency is tracked in #379.

> I wish new chats were displayed with the most recent first (highest) rather than oldest first in the rag ui.

That is reasonable. Would you mind creating an issue for this?

> I also wish uploading a document was optional (no RAG) though I realize you might want to specialize in RAG specifically.

Yes, RAG is the focus and we are not going to provide an interface just around LLMs. There are plenty of tools out there that do that.

**Comment by Adam-D-Lewis** [comment]
> I think rich uses a best effort approach to display these. SO either I'm wrong or your terminal doesn't allow any symbols whatsoever. In any case, we should probably look into it. Would you mind creating an issue for this?

Thanks for the response.  For a little more detail, it was working in the vs code terminal, but not through Konsole.  It appears this is a known issue and I was able to fix it by following the steps in https://www.reddit.com/r/archlinux/comments/o1n6fd/tip_for_anyone_struggling_to_get_emojis_to/.

I'll open an issue for the chat ordering.
